### ISSUE TITLE
diff for k8s secrets using CLI

### ISSUE THREAD

**Issue by marcb1** [argoproj/argo-cd#1442]
using argocd `v0.12.0+cd4bb25`, we have some kubernetes secrets in git in the form of a yaml files, for example a webhook secret:
```
apiVersion: v1
data:
  WEBHOOK_SECRET: test
kind: Secret
metadata:
  creationTimestamp: null
  name: mrmeeseeks-production
```

The web UI does not show any diffs on this, however running `argocd app [app name] diff --local=.`, shows a diff like so:
```
<   WEBHOOK_SECRET: '*********'
>   CERTIFICATE: test
```
Is there better way of doing this, either have the CLI tool ignore `****` secrets, or if there's a better way of managing secrets let me know. I realize having them in git is not ideal, but there are some secrets that are not private keys but just random strings.

**Comment by alexmt** [comment]
I agree. CLI just don't have access to secret data and cannot compare secrets with local manifests. I think secrets should be just ignored.

**Comment by alexmt** [comment]
This is implemented by #1459 don't compare secrets, since argo-cd doesn't have access to their data

**Comment by aweis89** [comment]
I think it would be ideal if the CLI was able to indicate whether a secret is being changed. Perhaps we can expose a specific endpoint that accepts a secret and responds `true/false` whether there's a diff with the live secret. That way the CLI can indicate whether a secret is being changed without needing to expose actual secret values
